node/node.go

```diff
@@ -636,17 +646,66 @@ func (n *Node) OnStart() error {
 		return ErrDialPeers{Err: err}
 	}
 
-	// Run state sync
+	// Start statesync.
 	if n.stateSync {
 		bcR, ok := n.bcReactor.(blockSyncReactor)
 		if !ok {
 			return ErrSwitchStateSync
 		}
-		err := startStateSync(n.stateSyncReactor, bcR, n.stateSyncProvider,
-			n.config.StateSync, n.stateStore, n.blockStore, n.stateSyncGenesis, n.config.Storage.ExperimentalKeyLayout)
+
+		stateCh, errc, err := startStateSync(n.stateSyncReactor, n.stateSyncProvider,
+			n.config.StateSync, n.stateStore, n.blockStore, n.stateSyncState, n.config.Storage.ExperimentalKeyLayout)
 		if err != nil {
 			return ErrStartStateSync{Err: err}
 		}
+
+		// Wait until statesync is over or errors, and start blocksync in a separate goroutine.
+		go func() {
```

Comment:
I was a bit unsure whether the parallelism here and the channels open a way for introducing errors but I can't think of a scenario atm.
The things I was trying to understand:
1. Can we have `state` returned on the `stateCh` before the Syncing function sets the syncing flag to false (which should not happen after we switch to blocksync) - but I think it is not possible. 
2. Could the note somehow setup wrongly, but again I am not seeing it with this code. 

Writing this just in case something clicks on your end when reading this :D

Reply:
I will change it to wait. I just noticed that we're sleeping in `Node.Start` 😄

Reply:
Where is the sleep?

Reply:
But is this not only an exception and an error case while in the regular case tehre is no sleep?

Reply:
True. Still avoiding extra goroutines gives app developers more control over the `Start` function.

Reply:
Tbh I would avoid changing the behaviour that was there for some years w/o being sure it hsa no implications. For example, someone expecting `node.Start()` to return w/o statesync might be doing other things that could facilitate StateSync execution. 

So I would put statesync in a separate go routine as it was.